src/VCR/Storage/AbstractStorage.php

```diff
@@ -50,12 +51,23 @@
     /**
      * Creates a new file store.
      *
+     * If the cassetteName contains PATH_SEPARATORs, subfolders of the
+     * cassettePath are autocreated when not existing.
+     *
      * @param string $cassettePath Path to the cassette directory.
-     * @param string $cassetteName Path to a file, will be created if not existing.
+     * @param string $cassetteName Path to the cassette file, relative to the path.
      */
     public function __construct($cassettePath, $cassetteName, $defaultContent = '[]')
     {
-        $file = $cassettePath . DIRECTORY_SEPARATOR . $cassetteName;
+        if (!is_dir($cassettePath)) {
+            throw new VCRException(sprintf('Cassette path "%s" is not existing or not a directory', $cassettePath), 0);
```

Comment:
To me `Assertion::directory($cassettePath, "Cassette path "'{$cassettePath}" is not existing or not a directory");` would be most understandable, just like the other assertions a bit further down in the constructor. Also it throws the right exception.

The same check is btw done [here](https://github.com/php-vcr/php-vcr/blob/master/src/VCR/Configuration.php#L381) with a very descriptive error message intended for end users. But it certainly doesn't hurt to do it here as well.